src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs

```diff
@@ -127,15 +126,26 @@ private IODataModelAdapter LoadModelAdapter(ITypeCache typeCache, string modelAd
             }
         }
 
-        private Assembly LoadAdapterAssembly(string modelAdapterAssemblyName)
+        private Type LoadType(string assemblyName, string typeName)
         {
-            var assemblyName = new AssemblyName(modelAdapterAssemblyName);
-            return Assembly.Load(assemblyName);
+            try
+            {
+                var assemblyNameObj = new AssemblyName(assemblyName);
+                return Assembly.Load(assemblyNameObj).GetType(typeName);
+            }catch(FileNotFoundException fileNotFoundException)
+            {
+                var type = Assembly.GetExecutingAssembly().GetType(typeName);
```

Comment:
Why not `Type.GetType(typeName)` ?

Reply:
Indeed, it's better. I do not know this shortcut thank you.